erpnext_shopify/sync_customers.py

```diff
@@ -0,0 +1,74 @@
+import frappe
+from frappe import _
+from .utils import get_address_type
+from .exceptions import ShopifyError
+from .shopify_requests import get_shopify_customers, post_request
+
+def sync_customers():
+	sync_shopify_customers()
+	sync_erp_customers()
+
+def sync_shopify_customers():
+	for customer in get_shopify_customers():
+		if not frappe.db.get_value("Customer", {"shopify_id": customer.get('id')}, "name"):
+			create_customer(customer)
+
+def create_customer(customer):
+	erp_cust = None
+	cust_name = (customer.get("first_name") + " " + (customer.get("last_name") and  customer.get("last_name") or ""))\
+		if customer.get("first_name") else customer.get("email")
+
+	try:
+		erp_cust = frappe.get_doc({
+			"doctype": "Customer",
+			"name": customer.get("id"),
+			"customer_name" : cust_name,
+			"shopify_id": customer.get("id"),
+			"customer_group": "Commercial",
+			"territory": "All Territories",
+			"customer_type": "Company"
+		}).insert()
+	except:
+		pass
+
+	if erp_cust:
+		create_customer_address(erp_cust, customer)
+
+def create_customer_address(erp_cust, customer):
+	for i, address in enumerate(customer.get("addresses")):
+		frappe.get_doc({
+			"doctype": "Address",
+			"address_title": erp_cust.customer_name,
+			"address_type": get_address_type(i),
+			"address_line1": address.get("address1") or "Address 1",
+			"address_line2": address.get("address2"),
+			"city": address.get("city") or "City",
+			"state": address.get("province"),
+			"pincode": address.get("zip"),
+			"country": address.get("country"),
+			"phone": address.get("phone"),
+			"email_id": customer.get("email"),
+			"customer": erp_cust.name,
+			"customer_name":  erp_cust.customer_name
+		}).insert()
```

Comment:
Does a shopify customer address have something we can store like `shopify_address_id`? We will need to sync it.